SU2_CFD/src/output/filewriter/CSTLFileWriter.cpp

```diff
@@ -0,0 +1,328 @@
+                CFileWriter(std::move(fields),
+                            std::move(fileName),
+                            dataSorter,
+                            fileExt,
+                            nDim){}
+
+
+CSTLFileWriter::~CSTLFileWriter(){}
+
+
+void CSTLFileWriter::Write_Data(){
+
+  /*--- This Write_Data routine has 3 major parts:
+    1. Prerequisite info: The parallel data-sorter distributes nodes of the primal mesh onto the processes
+    (i.e. this step is only important for parallel excecution) and not elements. For the STL file the connectivity
+    of the elements is broken/lost especially on the node-borders between processes. This information is
+    re-processed in this first (cumbersome) step.
+    2. The coordinate data for each node in a Triangle is successively written into a local array.
+    Quadrilateral surface elements are split into 2 triangles. All local arrays are gathered on the MASTER_NODE.
+    3. The MASTER_NODE only writes the data into a .stl file.
+  ---*/
+
+  /*--- 1. Re-process the element connectivity information and store that appropriatly such that
+    this information can be accessed in step 2---*/
+
+  unsigned long nParallel_Tria = dataSorter->GetnElem(TRIANGLE),
+                nParallel_Quad = dataSorter->GetnElem(QUADRILATERAL);
+  unsigned short iVar;
+
+  num_nodes_to_receive.resize(size, 0);
+  values_to_receive_displacements.resize(size);
+  halo_nodes.clear();
+  sorted_halo_nodes.clear();
+
+  /* We output a single, partitioned zone where each rank outputs one partition. */
+  vector<int32_t> partition_owners;
+  partition_owners.reserve(size);
+  for (int32_t iRank = 0; iRank < size; ++iRank)
+    partition_owners.push_back(iRank);
+
+  /* Gather a list of nodes we refer to but are not outputting. */
+  for (unsigned long i = 0; i < nParallel_Tria * N_POINTS_TRIANGLE; ++i)
+    if (dataSorter->FindProcessor(dataSorter->GetElem_Connectivity(TRIANGLE, 0, i)-1) != rank)
+      halo_nodes.insert(dataSorter->GetElem_Connectivity(TRIANGLE, 0, i)-1);
+
+  for (unsigned long i = 0; i < nParallel_Quad * N_POINTS_QUADRILATERAL; ++i)
+    if (dataSorter->FindProcessor(dataSorter->GetElem_Connectivity(QUADRILATERAL, 0, i)-1) != rank)
+      halo_nodes.insert(dataSorter->GetElem_Connectivity(QUADRILATERAL, 0, i)-1);
+
+
+  /* Sorted list of halo nodes for this MPI rank. */
+  sorted_halo_nodes.assign(halo_nodes.begin(), halo_nodes.end());
+
+  /*--- We effectively tack the halo nodes onto the end of the node list for this partition.
+    TecIO will later replace them with references to nodes in neighboring partitions. */
+  num_halo_nodes = sorted_halo_nodes.size();
+  vector<int64_t> halo_node_local_numbers(max((size_t)1, num_halo_nodes)); /* Min size 1 to avoid crashes when we access these vectors below. */
+  vector<int32_t> neighbor_partitions(max((size_t)1, num_halo_nodes));
+  vector<int64_t> neighbor_nodes(max((size_t)1, num_halo_nodes));

```

Comment:
I've seen, int, unsigned long, size_t, int32, int64, size_t, ... is this really needed? or can we make do with just a couple types (just because the static_casts everywhere are ugly).

Reply:
This is copied TecIO code, so it was probably important for tecplot output. I removed all size_t, int32, int64 and replaced them by unsigned long, int, long. I got rid of a few casts, but not all.

If you want to take a second look, the code is now in a seperate ReprocessElementConnectivity routine